docs/kubernetes/crds/agentendpoint.mdx

```diff
@@ -0,0 +1,389 @@
+---
+title: AgentEndpoint
+pagination_next: kubernetes/crds/cloudendpoint
+---
+
+## AgentEndpoint Custom Resource
+
+### **apiVersion:** `ngrok.k8s.ngrok.com/v1alpha1`
+
+### **kind:** `AgentEndpoint`
+
+Agent endpoints are ephemeral endpoints tied to the lifetime of the agent. When used with the ngrok Kubernetes operator,
+this means that the Agent Endpoints are tied to the operator's `operator-agent` pod (which is deployed by default when installing the operator).
+The operator will manage, create, and delete agent endpoints for you according to the configuration of the `AgentEndpoint` custom resources you create.
+So long as at least one instance of the `operator-agent` pod is running, your agent endpoints will be available. You may occasionally notice
+the IDs of Agent Endpoints managed by the operator change if the operator pods restart, this will not halt traffic through your agent endpoints unless all of the operator pods have stopped.
```

Comment:
```suggestion
the IDs of Agent Endpoints managed by the operator change if the operator pods restart, this will not halt traffic through your Agent Endpoints unless all of the operator pods have stopped.
```